src/gtest/test_checktransaction.cpp

```diff
@@ -1291,10 +1286,17 @@ TEST(ChecktransactionTests, HeartwoodEnforcesSaplingRulesOnShieldedCoinbase) {
     MockCValidationState state;
     EXPECT_TRUE(CheckTransactionWithoutProofVerification(tx, state));
 
-    // Coinbase transaction does not pass contextual checks, as bindingSig
+    // Coinbase transaction should pass contextual checks.
+    EXPECT_TRUE(ContextualCheckTransaction(tx, state, chainparams, 10, 57));
+
+    auto orchardAuth = orchard::AuthValidator::Disabled();
+    auto heartwoodBranchId = NetworkUpgradeInfo[Consensus::UPGRADE_HEARTWOOD].nBranchId;
+
+    // Coinbase transaction does not pass shielded input checks, as bindingSig
     // consensus rule is enforced.
     EXPECT_CALL(state, DoS(100, false, REJECT_INVALID, "bad-txns-sapling-binding-signature-invalid", false, "")).Times(1);
```

Comment:
Strictly speaking, expecting a call to `DoS` does not ensure that consensus checks fail. Typically, a failure looks like `return state.DoS(...)` where `DoS` returns false, but it's returning `false` that is most important. The call to `DoS` only indicates what failed, and an erroneous check might call it without returning `false`.